src/skin.cpp

```diff
@@ -648,7 +648,7 @@ void SkinInfillAreaComputation::generateTopAndBottomMostSkinSurfaces(SliceLayerP
 
     for (SkinPart& skin_part : part.skin_parts) {
         Polygons no_air_above = generateNoAirAbove(part, 1);
-        skin_part.top_most_surface_fill = skin_part.skin_fill.difference(no_air_above);
+        skin_part.top_most_surface_fill = skin_part.roofing_fill.difference(no_air_above);
```

Comment:
Not sure if this is expected behavior, but if we set `top surface skin layers = 0` then we do **not** get retractions anymore on the top surface layer. If this is intended behavior then disregard my comment. If this is not intended behavior and we do want retraction on the layer that is exposed to air disregarding if it is a `top surface skin layer` or not then I expect we should apply the following change.
```suggestion
        skin_part.top_most_surface_fill = skin_part.outline.difference(no_air_above);
```